app/controllers/concerns/metadata.rb

```diff
@@ -62,11 +62,16 @@ def last_offset(total)
 
     private
 
+    def includes_link
+      "&include=#{params[:include]}" if params[:include].present?
+    end
+
     def base_link
       base_url = "#{path_prefix}/#{Settings.app_name}"\
                  "/#{controller_name}"
       base = "#{base_url}?limit=#{pagination_limit}"
       base += "&search=#{params[:search]}" if params[:search].present?
+      base += includes_link.to_s
```

Comment:
Are any of those properly URL encoded? How about passing quotes or <> chars (esp. within the search param)?

Reply:
Yes, nothing is encoded, but is this an issue? I'm not sure I understand the XSS risk.

```
    "links": {
        "first": "/api/compliance/systems?limit=10&search=os_major_version < 9&offset=1",
        "last": "/api/compliance/systems?limit=10&search=os_major_version < 9&offset=1"
    },
```

Reply:
The risk is only potential. Something like: `<a href="{{ links.last }}">`. Framework should handle it, however our output could be more defensive in this regard.

Reply:
I've reported it as RHICOMPL-1465